src/Microsoft.DncEng.SecretManager/SecretTypes/AzureStorageSasSecretType.cs

```diff
@@ -0,0 +1,23 @@
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.WindowsAzure.Storage;
+
+namespace Microsoft.DncEng.SecretManager.SecretTypes
+{
+    public abstract class AzureStorageSasSecretType : SecretType
+    {
+        private readonly string _connectionStringName;
+
+        protected AzureStorageSasSecretType(IReadOnlyDictionary<string, string> parameters) : base(parameters)
+        {
+            ReadRequiredParameter("connectionStringName", ref _connectionStringName);
+        }
+
+        protected async Task<CloudStorageAccount> ConnectToAccount(RotationContext context, CancellationToken cancellationToken)
+        {
+            var connectionString = await context.GetSecretValue(_connectionStringName);
```

Comment:
Does the "context" always point to the same vault? I can see scenarios where we'd let people have SAS tokens to specific things, but wouldn't want them to have access to the AccessKey values (since those are STUPID powerful), so those would be in a different, more restricted vault.

Reply:
Currently it only points to one vault, I like the manifest being self contained. For "cross-vault" SAS tokens like that I could add another mode to the secret type that reads the connection string from another vault, but that would require knowledge about cross-manifest dependencies to prevent invalidating the SAS token when the connection string gets rotated. In normal operation that wouldn't happen given the current rotation intervals, but if we force-rotate for some reason that other SAS would get broken.

Reply:
Makes sense. :-)